docs/ARCHITECTURE.md

```diff
@@ -52,38 +54,36 @@ output on to the next one.
 ```
 
 ### Shutdown
-To ensure that all tasks shutdown on completion (or failure) of the 
-simulation, we relay on the following: 
-1. [Triggered](https://docs.rs/triggered/latest/triggered): a `Trigger` 
-  that can be used to inform threads that it's time to shut down, and 
-  a `Listener` that propagates this signal.
-2. Channel mechanics: if all of the senders for a channel have been 
-  dropped (in our context, all of the producers have exited) then the 
-  corresponding receiving channel will error out on receiving. Likewise, 
-  if the receiving channel is dropped, the sending channels will error 
-  out.
-
-All events are handled in a `tokio::select` to allow waiting on 
-multiple asynchronous tasks at once. These selects should be `biased` 
-on the exit case (ie, the `Listener` being triggered) so that we 
+
+To ensure that all tasks shutdown on completion (or failure) of the
+simulation, we relay on the following:
+
+1. [Triggered](https://docs.rs/triggered/latest/triggered): a `Trigger`
+   that can be used to inform threads that it's time to shut down, and
+   a `Listener` that propagates this signal.
+2. The (`Trigger`, `Listener`) pair are used with channels: if a channel errors out across `send()` or `receive()`, shutdown is triggered. There is no reliance on channel mechanics, i.e. errors generated when all senders are and/or a receiver is dropped.
+
+All events are handled in a `tokio::select` to allow waiting on
+multiple asynchronous tasks at once. These selects should be `biased`
+on the exit case (ie, the `Listener` being triggered) so that we
 prioritize exit above generating more events.
 
-Practically, this means that we're handling the various shutdown 
-scenarios in the following way: 
+Practically, this means that we're handling the various shutdown
+scenarios in the following way:
+
+1. Consumer error:
+
+- Receiving channel is dropped on consumer exit.
```

Comment:
We shouldn't be relying on this now? We can probably cut this whole section now that we're just using listener/trigger everywhere.

Reply:
This comment still applies, IMO we can just delete the whole paragraph under "practically, this means that we're handling"...